neo/d3xp/Player.cpp

```diff
@@ -5694,20 +5694,27 @@ void idPlayer::Weapon_Combat()
 			
 			if( weapon.GetEntity()->IsHolstered() )
 			{
-				assert( idealWeapon.Get() >= 0 );
-				assert( idealWeapon.Get() < MAX_WEAPONS );
-				
-				if( currentWeapon != weapon_pda && !spawnArgs.GetBool( va( "weapon%d_toggle", currentWeapon ) ) )
+				if ( idealWeapon.Get() == 0 )
```

Comment:
Hmm idealWeapon being 0 should be ok - we have def_weapon0 and def_weapon1 in def/player.def, so the indices 0 and 1 should be valid.
We should make sure idealWeapon is not -1 though.
We'll have to investigate this problem (where does it crash/freeze exactly and why? what's the value of idealWeapon there? where was that value set?) properly.